src/modules/Navigation/index.tsx

```diff
@@ -63,6 +67,25 @@ const Navigation = ({ children }: NavigationProps): JSX.Element => {
         };
     });
 
+    useEffect(() => {
```

Comment:
if the child component says it is loading, wait 75ms, and if it is still loading, start the loading spinner. After the child says it has loaded, wait another 750ms and start the loading spinner's fade out during this time.